### ISSUE TITLE
There are no commands defined in the "devise" namespace.

### ISSUE THREAD

**Issue by CSCG** [devisephp/cms#11]
Hey all,

Very, VERY interesting project, and it appears to fit my usecase nicely. However, the installation procedure is a little finicky compared to others - not sure if I'm missing something here, but I've gone ahead and followed the install guide on which it instructs solely to run the artisan devise:install command, on which my artisan install in stock Laravel certainly can't find any commands...what am I missing here? Do I need to download some further files? This may seem pedantic and silly to those developing the project, but the documentation doesn't speak anything in this regard.

[InvalidArgumentException]

There are no commands defined in the "devise" namespace.

**Comment by pistonhonda** [comment]
CSCG, that's not silly. I think we made a few assumptions, when writing that section, and need to clarify.

We'll get that updated ASAP.

However if you've got a fresh Laravel project started, you can follow the "Installing Devise On An Existing Project" steps 1 & 2. Then you'll then see the commands in the "devise" namespace. After that you can run the devise:install command.

Sorry for any confusion. Let us know how your project goes. We love feed back :)

**Comment by CSCG** [comment]
@pistonhonda I heavily appreciate the swift reply. I went ahead and cloned the bootstrap'd project in the interest of starting development, and the install launched right away upon navigation to public/, which is fantastic! However...midway through the install the console began throwing some issues:

http://domain.us/packages/devisephp/cms/js/app/admin/main.js Failed to load resource: the server responded with a status of 404 (Not Found)
devise.min.js:1 Uncaught Error: Script error for: app/admin/main
http://requirejs.org/docs/errors.html#scripterror
http://domain.us/packages/devisephp/cms/js/app/admin/admin.js Failed to load resource: the server responded with a status of 404 (Not Found)
devise.min.js:1 Uncaught Error: Script error for: app/admin/admin
http://requirejs.org/docs/errors.html#scripterror

and then the worker successfully recognized an issue occurred, and pulled the install to a stop-status with the below error, not sure really what's the error here beyond some missing javascript files. I'm gonna go ahead and attempt the alternate method you replied with, I'll report back with any information:

Whoops, looks like something went wrong.

1/1
FatalErrorException in Filesystem.php line 158:
Maximum execution time of 30 seconds exceeded
in Filesystem.php line 158

**Comment by CSCG** [comment]
Just an update -- it seems like it's looking one level too low. The correct call URL should be: http://domain.us/public/packages/devisephp/cms/js/app/admin/main.js - I'm not really sure if I'm running a nontraditional URL structure here, but I don't believe so!

**Comment by pistonhonda** [comment]
@CSCG Yeah, for Laravel that is a bit "nontraditional" the "/public" directory should resolve as "/". So Devise is assuming the main.js path is http://domain.us/packages/devisephp/cms/js/app/admin/main.js

Are you working with Homestead?

**Comment by CSCG** [comment]
@pistonhonda gotcha. I'm actually like 60 days deep in a lonnnnnnnnng argument over structure of application and URIs with TO over in the Lumen repo, that would make sense...I really hate that restriction, based entirely around the .env config structure and it's dependent permissions I think.

I'm actually working on a Dreamhost account with limited SSH - this is the main host environment for many of my clients, which is why I've been fighting this issue across repos for quite some time...technically, it IS possible to set up public to resolve correctly using the internal FTP Dreamhost structure, but everytime they update the server or Apache, the .htaccess borks itself and everything for all clients fall apart.

So, in conclusion, let's go ahead and close this issue :+1: I apologize for all the text - I'm gonna set up a DigitalOcean droplet and test this out in there instead, perhaps. 

Thanks again, you'll see more of me soon!

**Comment by dolbex** [comment]
Just FYI - we have updated the docs appropriately - thanks for the persistence @CSCG

**Comment by CSCG** [comment]
@dolbex you guys rule. I'll take another look into rolling devisephp towards our UAT environment over the next cycle, scheduled in about ten days!

cheers!
